freqtrade/rpc/api_server/ws/channel.py

```diff
@@ -0,0 +1,169 @@
+import logging
+from threading import RLock
+from typing import List, Type
+
+from freqtrade.rpc.api_server.ws.proxy import WebSocketProxy
+from freqtrade.rpc.api_server.ws.serializer import ORJSONWebSocketSerializer, WebSocketSerializer
+from freqtrade.rpc.api_server.ws.types import WebSocketType
+
+
+logger = logging.getLogger(__name__)
+
+
+class WebSocketChannel:
+    """
+    Object to help facilitate managing a websocket connection
+    """
+
+    def __init__(
+        self,
+        websocket: WebSocketType,
+        serializer_cls: Type[WebSocketSerializer] = ORJSONWebSocketSerializer
```

Comment:
we're using json now as default, right? 

I agree that should be the default for a new connection - but i guess part of the initializtion can be the client sending "i'd like msgpack instead" - so the decision which protocol to use is up to the client.

Reply:
So nothing is actually passed to the `serializer_cls` parameter so it just uses ORJSON as default right now. We want this configurable eventually though.

Reply:
no - the client takes that decision, otherwise it'll have to be one method only.

you can always have alternative consumers connecting to it, which don't support msgpack - if the producer decides "i use msgpack" - it'd break all other consumers.

So i see 2 options:
* if you wanna support multiple methods - the consumer decides
* if you wanna support one method - it'll be json.

Reply:
So I don't see a way to support having the consumer decide the serialization, as it would require passing a message through the websocket to tell the producer, and by that time the serialization of the websocket is already decided. In that case I suppose we will just support ORJSON

Reply:
well my idea was to start with json and start it uppon request. What's the problems you see with that?

Reply:
I think that would add some unnecessary complexities, however that could be something we revisit in the future. As of right now though that's not essential.